File: apps/web/src/api/enrollment.ts
import client from './client';

export type EnrollmentStatus = 'pending' | 'reviewing' | 'approved' | 'rejected';

export interface EnrollmentData {
  id: string;
  tenant_id: string;
  parent_first_name: string;
  parent_last_name: string;
  parent_email: string;
  parent_phone: string | null;
  child_first_name: string;
  child_last_name: string;
  child_date_of_birth: string | null;
  child_gender: string | null;
  requested_group_id: string | null;
  status: EnrollmentStatus;
  admin_notes: string | null;
  reviewed_by: string | null;
  reviewed_at: string | null;
  documents: string[];
  student_user_id: string | null;
  submitted_by: string | null;
  created_at: string;
  updated_at: string;
}

export interface EnrollmentCreatePayload {
  parent_first_name: string;
  parent_last_name: string;
  parent_email: string;
  parent_phone?: string;
  child_first_name: string;
  child_last_name: string;
  child_date_of_birth?: string;
  child_gender?: string;
  requested_group_id?: string;
  documents?: string[];
}

export interface EnrollmentReviewPayload {
  status: 'reviewing' | 'approved' | 'rejected';
  admin_notes?: string;
}

export function listEnrollments(status?: EnrollmentStatus) {
  return client.get<EnrollmentData[]>('/v1/enrollment', { params: status ? { status } : {} });
}

export function getMyEnrollments() {
  return client.get<EnrollmentData[]>('/v1/enrollment/my');
}

export function submitEnrollment(data: EnrollmentCreatePayload) {
  return client.post<EnrollmentData>('/v1/enrollment', data);
}

export function reviewEnrollment(id: string, data: EnrollmentReviewPayload) {
  return client.patch<EnrollmentData>(`/v1/enrollment/${id}/review`, data);
}

File: apps/api/app/db/models/enrollment.py
import uuid
from datetime import datetime

from sqlalchemy import Column, DateTime, ForeignKey, Integer, String, Text
from sqlalchemy.dialects.postgresql import JSONB, UUID
from sqlalchemy.orm import relationship

from app.db.base import Base, TenantMixin


class EnrollmentRequest(Base, TenantMixin):
    __tablename__ = "enrollment_requests"

    # Parent/guardian info
    parent_first_name = Column(String(100), nullable=False)
    parent_last_name = Column(String(100), nullable=False)
    parent_email = Column(String(255), nullable=False)
    parent_phone = Column(String(50), nullable=True)

    # Child info
    child_first_name = Column(String(100), nullable=False)
    child_last_name = Column(String(100), nullable=False)
    child_date_of_birth = Column(DateTime, nullable=True)
    child_gender = Column(String(20), nullable=True)  # male|female|other|undisclosed

    # Requested group (class)
    requested_group_id = Column(UUID(as_uuid=True), ForeignKey("groups.id"), nullable=True)

    # Status workflow
    status = Column(String(20), default="pending", nullable=False)
    # pending | reviewing | approved | rejected

    # Admin notes
    admin_notes = Column(Text, nullable=True)
    reviewed_by = Column(UUID(as_uuid=True), ForeignKey("users.id"), nullable=True)
    reviewed_at = Column(DateTime, nullable=True)

    # Uploaded documents (list of file IDs or S3 keys)
    documents = Column(JSONB, default=list)

    # On approval: reference to created student user
    student_user_id = Column(UUID(as_uuid=True), ForeignKey("users.id"), nullable=True)

    # Submitter user (if parent is already a user)
    submitted_by = Column(UUID(as_uuid=True), ForeignKey("users.id"), nullable=True)
    # Payment gate (optional): enrollment confirmed after invoice paid
    invoice_id = Column(UUID(as_uuid=True), nullable=True)
    payment_minimum_cents = Column(Integer, nullable=False, default=0)  # 0 = full required

